// react-native-notifier/lib/commonjs/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DEFAULT_SWIPE_ENABLED = exports.SWIPE_ANIMATION_DURATION = exports.SWIPE_PIXELS_TO_CLOSE = exports.DEFAULT_ANIMATION_DURATION = exports.DEFAULT_DURATION = exports.DEFAULT_COMPONENT_HEIGHT = exports.MAX_TRANSLATE_Y = exports.MIN_TRANSLATE_Y = void 0;
const MIN_TRANSLATE_Y = -1000;
exports.MIN_TRANSLATE_Y = MIN_TRANSLATE_Y;
const MAX_TRANSLATE_Y = 0;
exports.MAX_TRANSLATE_Y = MAX_TRANSLATE_Y;
const DEFAULT_COMPONENT_HEIGHT = 200;
exports.DEFAULT_COMPONENT_HEIGHT = DEFAULT_COMPONENT_HEIGHT;
const DEFAULT_DURATION = 3000;
exports.DEFAULT_DURATION = DEFAULT_DURATION;
const DEFAULT_ANIMATION_DURATION = 300;
exports.DEFAULT_ANIMATION_DURATION = DEFAULT_ANIMATION_DURATION;
const SWIPE_PIXELS_TO_CLOSE = 20;
exports.SWIPE_PIXELS_TO_CLOSE = SWIPE_PIXELS_TO_CLOSE;
const SWIPE_ANIMATION_DURATION = 200;
exports.SWIPE_ANIMATION_DURATION = SWIPE_ANIMATION_DURATION;
const DEFAULT_SWIPE_ENABLED = true;
exports.DEFAULT_SWIPE_ENABLED = DEFAULT_SWIPE_ENABLED;
