// react-native-notifier/lib/module/components/Alert.js
import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import SafeContainer from './SafeContainer';
const bgColors = {
  error: '#E03737',
  warn: '#FFAC00',
  info: '#007BFF',
  success: '#00B104'
};
const s = StyleSheet.create({
  container: {
    margin: 10
  },
  title: {
    fontWeight: 'bold',
    textAlign: 'center',
    fontSize: 15,
    lineHeight: 22,
    color: 'white'
  },
  description: {
    textAlign: 'center',
    fontSize: 14,
    lineHeight: 22,
    color: 'white'
  }
});

const AlertComponent = ({
  title,
  titleStyle,
  description,
  descriptionStyle,
  alertType = 'success',
  backgroundColor,
  textColor,
  ContainerComponent,
  maxTitleLines,
  maxDescriptionLines
}) => {
  const Container = ContainerComponent !== null && ContainerComponent !== void 0 ? ContainerComponent : SafeContainer;
  const textStyle = textColor ? {
    color: textColor
  } : null;
  return /*#__PURE__*/React.createElement(Container, {
    style: {
      backgroundColor: backgroundColor || bgColors[alertType]
    }
  }, /*#__PURE__*/React.createElement(View, {
    style: s.container
  }, !!title && /*#__PURE__*/React.createElement(Text, {
    style: [s.title, textStyle, titleStyle],
    numberOfLines: maxTitleLines
  }, title), !!description && /*#__PURE__*/React.createElement(Text, {
    style: [s.description, textStyle, descriptionStyle],
    numberOfLines: maxDescriptionLines
  }, description)));
};

export default AlertComponent;
