// react-native-notifier/lib/module/components/Notification.js
import React from 'react';
import { StyleSheet, View, Text, Image } from 'react-native';
import SafeContainer from './SafeContainer';
const s = StyleSheet.create({
  container: {
    shadowColor: '#000000',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 10,
    backgroundColor: '#ffffff',
    borderRadius: 5,
    marginTop: 10,
    marginHorizontal: 10,
    paddingVertical: 10,
    paddingHorizontal: 10,
    flexDirection: 'row',
    alignItems: 'center'
  },
  image: {
    alignItems: 'center',
    marginRight: 5,
    borderRadius: 5,
    height: 45,
    width: 45
  },
  content: {
    flex: 1,
    marginHorizontal: 5
  },
  title: {
    fontWeight: 'bold',
    fontSize: 15,
    lineHeight: 25,
    color: '#007BFF'
  },
  description: {
    fontSize: 14,
    lineHeight: 20,
    color: '#444444'
  }
});

const NotificationComponent = ({
  title,
  titleStyle,
  description,
  descriptionStyle,
  imageSource,
  imageStyle,
  ContainerComponent,
  maxTitleLines,
  maxDescriptionLines,
  containerStyle
}) => {
  const Container = ContainerComponent !== null && ContainerComponent !== void 0 ? ContainerComponent : SafeContainer;
  return /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(View, {
    style: [s.container, containerStyle]
  }, !!imageSource && /*#__PURE__*/React.createElement(Image, {
    style: [s.image, imageStyle],
    source: imageSource
  }), /*#__PURE__*/React.createElement(View, {
    style: s.content
  }, !!title && /*#__PURE__*/React.createElement(Text, {
    style: [s.title, titleStyle],
    numberOfLines: maxTitleLines
  }, title), !!description && /*#__PURE__*/React.createElement(Text, {
    style: [s.description, descriptionStyle],
    numberOfLines: maxDescriptionLines
  }, description))));
};

export default NotificationComponent;
