// react-native-notifier/lib/module/components/SafeContainer.js
var _Platform$Version, _Platform$Version$toS, _Platform$Version$toS2;

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { StyleSheet, View, SafeAreaView, Dimensions, Platform } from 'react-native';
const isIOS10 = Platform.OS === 'ios' && ((_Platform$Version = Platform.Version) === null || _Platform$Version === void 0 ? void 0 : (_Platform$Version$toS = _Platform$Version.toString()) === null || _Platform$Version$toS === void 0 ? void 0 : (_Platform$Version$toS2 = _Platform$Version$toS.split('.')) === null || _Platform$Version$toS2 === void 0 ? void 0 : _Platform$Version$toS2[0]) === '10';
const s = StyleSheet.create({
  statusBarPadding: {
    paddingTop: 20
  }
});

// This component used instead of SafeAreaView on iOS 10 because of bug
// Details: https://github.com/seniv/react-native-notifier/issues/3
class StatusBarSpacer extends React.Component {
  constructor(props) {
    super(props);
    const {
      height,
      width
    } = Dimensions.get('window');
    this.state = {
      displayPadding: height > width
    };
    this.onSizeChange = this.onSizeChange.bind(this);
  }

  componentDidMount() {
    Dimensions.addEventListener('change', this.onSizeChange);
  }

  componentWillUnmount() {
    Dimensions.removeEventListener('change', this.onSizeChange);
  }

  onSizeChange({
    window
  }) {
    this.setState({
      displayPadding: window.height > window.width
    });
  }

  render() {
    const {
      style,
      ...props
    } = this.props;
    const {
      displayPadding
    } = this.state;
    return /*#__PURE__*/React.createElement(View, _extends({
      style: [displayPadding && s.statusBarPadding, style]
    }, props));
  }

}

export default isIOS10 ? StatusBarSpacer : SafeAreaView;
