// react-native-notifier/lib/commonjs/components/SafeContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

var _Platform$Version, _Platform$Version$toS, _Platform$Version$toS2;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const isIOS10 = _reactNative.Platform.OS === 'ios' && ((_Platform$Version = _reactNative.Platform.Version) === null || _Platform$Version === void 0 ? void 0 : (_Platform$Version$toS = _Platform$Version.toString()) === null || _Platform$Version$toS === void 0 ? void 0 : (_Platform$Version$toS2 = _Platform$Version$toS.split('.')) === null || _Platform$Version$toS2 === void 0 ? void 0 : _Platform$Version$toS2[0]) === '10';

const s = _reactNative.StyleSheet.create({
  statusBarPadding: {
    paddingTop: 20
  }
});

// This component used instead of SafeAreaView on iOS 10 because of bug
// Details: https://github.com/seniv/react-native-notifier/issues/3
class StatusBarSpacer extends _react.default.Component {
  constructor(props) {
    super(props);

    const {
      height,
      width
    } = _reactNative.Dimensions.get('window');

    this.state = {
      displayPadding: height > width
    };
    this.onSizeChange = this.onSizeChange.bind(this);
  }

  componentDidMount() {
    _reactNative.Dimensions.addEventListener('change', this.onSizeChange);
  }

  componentWillUnmount() {
    _reactNative.Dimensions.removeEventListener('change', this.onSizeChange);
  }

  onSizeChange({
    window
  }) {
    this.setState({
      displayPadding: window.height > window.width
    });
  }

  render() {
    const {
      style,
      ...props
    } = this.props;
    const {
      displayPadding
    } = this.state;
    return /*#__PURE__*/_react.default.createElement(_reactNative.View, _extends({
      style: [displayPadding && s.statusBarPadding, style]
    }, props));
  }

}

var _default = isIOS10 ? StatusBarSpacer : _reactNative.SafeAreaView;

exports.default = _default;
